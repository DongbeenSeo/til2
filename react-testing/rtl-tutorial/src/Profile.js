import React from 'react';

const Profile = ({ usename, name }) => {
  return (
    <div>
      <b>{usename}</b>&nbsp;
      <span>({name})</span>
    </div>
  );
};

export default Profile;
