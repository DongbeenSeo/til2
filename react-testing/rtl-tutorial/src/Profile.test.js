import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile/>', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile usename="dongbeen" name="서동빈" />);
    // const utils = render()
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the props correctly', () => {
    const utils = render(<Profile usename="dongbeen" name="서동빈" />);
    utils.getByText(/dongbeen/);
    utils.getByText(/서동빈/);
    utils.getByText(/서/);
  });
});
