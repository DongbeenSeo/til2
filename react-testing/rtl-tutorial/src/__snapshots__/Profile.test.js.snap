// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> matches snapshot 1`] = `
<div>
  <div>
    <b>
      dongbeen
    </b>
     
    <span>
      (
      서동빈
      )
    </span>
  </div>
</div>
`;
